torchvision/datasets/pascal.py

```diff
@@ -0,0 +1,170 @@
+from __future__ import print_function
+
+import errno
+import hashlib
+import os
+import sys
+import tarfile
+
+import torch.utils.data as data
+from PIL import Image
+
+from six.moves import urllib
+
+
+class PascalVOC(data.Dataset):
+    CLASSES = [
+        'background', 'aeroplane', 'bicycle', 'bird', 'boat', 'bottle', 'bus',
+        'car', 'cat', 'chair', 'cow', 'diningtable', 'dog', 'horse',
+        'motorbike', 'person', 'potted-plant', 'sheep', 'sofa', 'train',
+        'tv/monitor', 'ambigious'
+    ]
+
+    URL = "http://host.robots.ox.ac.uk/pascal/VOC/voc2012/VOCtrainval_11-May-2012.tar"
+    FILE = "VOCtrainval_11-May-2012.tar"
+    MD5 = '6cd6e144f989b92b3379bac3b3de84fd'
+    BASE_DIR = 'VOCdevkit/VOC2012'
+
+    def __init__(self,
+                 root,
+                 train=True,
+                 transform=None,
+                 target_transform=None,
+                 download=False):
+        self.root = root
+        voc_root = os.path.join(self.root, self.BASE_DIR)
+        mask_dir = os.path.join(voc_root, 'SegmentationClass')
+        image_dir = os.path.join(voc_root, 'JPEGImages')
+        self.transform = transform
+        self.target_transform = target_transform
+        self.train = train
+
+        if download:
+            self.download()
+
+        if not self._check_integrity():
+            raise RuntimeError('Dataset not found or corrupted.' +
+                               ' You can use download=True to download it')
+        # train/val/test splits are pre-cut
+        splits_dir = os.path.join(voc_root, 'ImageSets/Segmentation')
+        split_f = os.path.join(splits_dir, 'train.txt')
+        if not self.train:
+            split_f = os.path.join(splits_dir, ' trainval.txt')
+
+        self.images = []
+        self.masks = []
+        with open(os.path.join(split_f), "r") as lines:
+            for line in lines:
+                image = os.path.join(image_dir, line.rstrip('\n') + ".jpg")
+                mask = os.path.join(mask_dir, line.rstrip('\n') + ".png")
+                assert os.path.isfile(image)
+                assert os.path.isfile(mask)
+                self.images.append(image)
+                self.masks.append(mask)
+
+        assert (len(self.images) == len(self.masks))
+
+    def __getitem__(self, index):
+        img = Image.open(self.images[index]).convert('RGB')
+        target = Image.open(self.masks[index]).convert('RGB')
```

Comment:
This makes the target be a RGB image, and the labels become triplets. I think it's better to remove the `.convert('RGB')` to get the labels in the range 0-21, so that we can directly use them for training.